### DISCUSSION TITLE
Add real-time drawing capability to the node

### DISCUSSION THREAD

**Discussion by daizhuo** [Comfy-Org/ComfyUI#647] [Ideas]
What The biggest obstacle preventing me from using ComfyUI is its inability to directly draw on images like WebUI's imag2imag. This feature is very important. If you have to complete the drawing outside and then import it, it is very unfriendly.
Thank you!

**Comment by ltdrdata** [comment]
Do you meaning inpaint on img2img?
In that case..
mask editor feature on clipspace added recently.
And I have a plan to implement simple painting feature, too.

**Reply by daizhuo** [reply]
> Do you meaning inpaint on img2img? In that case.. mask editor feature on clipspace added recently. And I have a plan to implement simple painting feature, too.

Yes, img2img need this real-time drawing pad!
That's great!!! I'm looking forward it, can't to have it !
Thank you so much!

**Reply by LongToeBoy** [reply]
hey, i know there's lot's of work to be done besides it... but any progress on directly painting on images?

**Reply by ltdrdata** [reply]
This thread is quite old, and at the time, I was working with the ImageRefiner as an PoC application of the workflow component.

However, there are now many better alternatives, so I’m reconsidering my approach.

For now, take a look at something like Krita.

However, I agree that it would be better if ComfyUI supported it natively as a built-in feature lol.

**Reply by Psylenceo** [reply]
@LongToeBoy 
Have you either found an inpaint option or tried Krita with the comfyui plugin yet?
If you have not tried Krita with the plugin I also highly recommend it, especially with the external tooling nodes available within comfyui.
Here's a guide link that goes into some decent details about utilizing the nodes, https://docs.interstice.cloud/custom-graph/
By using the external nodes, you can open Krita and use it as an EXTREMELY powerful load image + masking + regional segmentation/IP/controlnet node that supports image recirculation, because you only need the node that lets you select image layers, and the output node. And because the output node returns the image to Krita as a new layer, you can make any adjustments you need then select the new layer and send it back through.
Also, the krita ai-diffusion plug-in's default txt2img / img2img lineart control net is insanely good! Plus having the ability to zoom in and actually add/remove fine details is incredibly nice.
